README.md
# vscode-theme-generator-quick-start

>Based almost entirely on [Tyriar's repo](https://github.com/Tyriar/vscode-theme-generator-quick-start).
>Updated to include *many* more color choices and a few options.

## Usage

**Clone and run for a quick way to get started with [vscode-theme-generator](https://github.com/skeptycal/vscode-theme-generator).**

## How to build the theme

You need [NodeJS](https://nodejs.org/en/) installed to run the generator.

```BASH
npm install
```

Make your changes to the colors in `index.ts` and hit `F5` to build the theme and launch the a new VS Code window with your theme available in the command palette (`ctrl`/`cmd+shift+p` > "Color Theme").

## Issues

- Dark backgrounds can cause certain text to appear 'invisible' since the chosen `background` and `foreground` defaults in the `index.ts` file are not replaced consistently

## Publishing

When you're ready to publish your theme just fill out the `"name"`, `"displayName", `"publisher"` and `"description"` values in the package.json and [publish it like any other extension](https://code.visualstudio.com/docs/extensions/publish-extension).


CHANGELOG.md
## 0.0.1

Initial release

## 0.0.2

Add larger color palette


more_colors.ts
// large standard color list
const colors = {
    black: '#000000',
    white: '#FFFFFF',
    red: '#FF0000',
    lime: '#00FF00',
    blue: '#0000FF',
    yellow: '#FFFF00',
    aqua: '#00FFFF',
    magenta: '#FF00FF',
    silver: '#C0C0C0',
    gray: '#808080',
    maroon: '#800000',
    olive: '#808000',
    green: '#008000',
    purple: '#800080',
    teal: '#008080',
    navy: '#000080',
    darkred: '#8B0000',
    brown: '#A52A2A',
    firebrick: '#B22222',
    crimson: '#DC143C',
    tomato: '#FF6347',
    coral: '#FF7F50',
    indianred: '#CD5C5C',
    lightcoral: '#F08080',
    darksalmon: '#E9967A',
    salmon: '#FA8072',
    lightsalmon: '#FFA07A',
    orangered: '#FF4500',
    darkorange: '#FF8C00',
    orange: '#FFA500',
    gold: '#FFD700',
    darkgoldenrod: '#B8860B',
    goldenrod: '#DAA520',
    palegoldenrod: '#EEE8AA',
    darkkhaki: '#BDB76B',
    khaki: '#F0E68C',
    yellowgreen: '#9ACD32',
    darkolivegreen: '#556B2F',
    olivedrab: '#6B8E23',
    lawngreen: '#7CFC00',
    chartreuse: '#7FFF00',
    greenyellow: '#ADFF2F',
    darkgreen: '#006400',
    forestgreen: '#228B22',
    limegreen: '#32CD32',
    lightgreen: '#90EE90',
    palegreen: '#98FB98',
    darkseagreen: '#8FBC8F',
    mediumspringgreen: '#00FA9A',
    springgreen: '#00FF7F',
    seagreen: '#2E8B57',
    mediumaquamarine: '#66CDAA',
    mediumseagreen: '#3CB371',
    lightseagreen: '#20B2AA',
    darkslategray: '#2F4F4F',
    darkcyan: '#008B8B',
    cyan: '#00FFFF',
    lightcyan: '#E0FFFF',
    darkturquoise: '#00CED1',
    turquoise: '#40E0D0',
    mediumturquoise: '#48D1CC',
    paleturquoise: '#AFEEEE',
    aquamarine: '#7FFFD4',
    powderblue: '#B0E0E6',
    cadetblue: '#5F9EA0',
    steelblue: '#4682B4',
    cornflowerblue: '#6495ED',
    deepskyblue: '#00BFFF',
    dodgerblue: '#1E90FF',
    lightblue: '#ADD8E6',
    skyblue: '#87CEEB',
    lightskyblue: '#87CEFA',
    midnightblue: '#191970',
    darkblue: '#00008B',
    mediumblue: '#0000CD',
    royalblue: '#4169E1',
    blueviolet: '#8A2BE2',
    indigo: '#4B0082',
    darkslateblue: '#483D8B',
    slateblue: '#6A5ACD',
    mediumslateblue: '#7B68EE',
    mediumpurple: '#9370DB',
    darkmagenta: '#8B008B',
    darkviolet: '#9400D3',
    darkorchid: '#9932CC',
    mediumorchid: '#BA55D3',
    thistle: '#D8BFD8',
    plum: '#DDA0DD',
    violet: '#EE82EE',
    fuchsia: '#FF00FF',
    orchid: '#DA70D6',
    mediumvioletred: '#C71585',
    palevioletred: '#DB7093',
    deeppink: '#FF1493',
    hotpink: '#FF69B4',
    lightpink: '#FFB6C1',
    pink: '#FFC0CB',
    antiquewhite: '#FAEBD7',
    beige: '#F5F5DC',
    bisque: '#FFE4C4',
    blanchedalmond: '#FFEBCD',
    wheat: '#F5DEB3',
    cornsilk: '#FFF8DC',
    lemonchiffon: '#FFFACD',
    lightgoldenrodyellow: '#FAFAD2',
    lightyellow: '#FFFFE0',
    saddlebrown: '#8B4513',
    sienna: '#A0522D',
    chocolate: '#D2691E',
    peru: '#CD853F',
    sandybrown: '#F4A460',
    burlywood: '#DEB887',
    tan: '#D2B48C',
    rosybrown: '#BC8F8F',
    moccasin: '#FFE4B5',
    navajowhite: '#FFDEAD',
    peachpuff: '#FFDAB9',
    mistyrose: '#FFE4E1',
    lavenderblush: '#FFF0F5',
    linen: '#FAF0E6',
    oldlace: '#FDF5E6',
    papayawhip: '#FFEFD5',
    seashell: '#FFF5EE',
    mintcream: '#F5FFFA',
    slategray: '#708090',
    lightslategray: '#778899',
    lightsteelblue: '#B0C4DE',
    lavender: '#E6E6FA',
    floralwhite: '#FFFAF0',
    aliceblue: '#F0F8FF',
    ghostwhite: '#F8F8FF',
    honeydew: '#F0FFF0',
    ivory: '#FFFFF0',
    azure: '#F0FFFF',
    snow: '#FFFAFA',
    dimgray: '#696969',
    darkgray: '#A9A9A9',
    lightgray: '#D3D3D3',
    gainsboro: '#DCDCDC',
    whitesmoke: '#F5F5F5'
}

index.ts
import * as path from 'path';
import { generateTheme, IColorSet } from 'vscode-theme-generator';

const themeName = 'Generated';
const colors = {
  black: '#000000',
  white: '#FFFFFF',
  red: '#FF0000',
  lime: '#00FF00',
  blue: '#0000FF',
  yellow: '#FFFF00',
  aqua: '#00FFFF',
  magenta: '#FF00FF',
  silver: '#C0C0C0',
  gray: '#808080',
  maroon: '#800000',
  olive: '#808000',
  green: '#008000',
  purple: '#800080',
  teal: '#008080',
  navy: '#000080',
  darkred: '#8B0000',
  brown: '#A52A2A',
  firebrick: '#B22222',
  crimson: '#DC143C',
  tomato: '#FF6347',
  coral: '#FF7F50',
  indianred: '#CD5C5C',
  lightcoral: '#F08080',
  darksalmon: '#E9967A',
  salmon: '#FA8072',
  lightsalmon: '#FFA07A',
  orangered: '#FF4500',
  darkorange: '#FF8C00',
  orange: '#FFA500',
  gold: '#FFD700',
  darkgoldenrod: '#B8860B',
  goldenrod: '#DAA520',
  palegoldenrod: '#EEE8AA',
  darkkhaki: '#BDB76B',
  khaki: '#F0E68C',
  yellowgreen: '#9ACD32',
  darkolivegreen: '#556B2F',
  olivedrab: '#6B8E23',
  lawngreen: '#7CFC00',
  chartreuse: '#7FFF00',
  greenyellow: '#ADFF2F',
  darkgreen: '#006400',
  forestgreen: '#228B22',
  limegreen: '#32CD32',
  lightgreen: '#90EE90',
  palegreen: '#98FB98',
  darkseagreen: '#8FBC8F',
  mediumspringgreen: '#00FA9A',
  springgreen: '#00FF7F',
  seagreen: '#2E8B57',
  mediumaquamarine: '#66CDAA',
  mediumseagreen: '#3CB371',
  lightseagreen: '#20B2AA',
  darkslategray: '#2F4F4F',
  darkcyan: '#008B8B',
  cyan: '#00FFFF',
  lightcyan: '#E0FFFF',
  darkturquoise: '#00CED1',
  turquoise: '#40E0D0',
  mediumturquoise: '#48D1CC',
  paleturquoise: '#AFEEEE',
  aquamarine: '#7FFFD4',
  powderblue: '#B0E0E6',
  cadetblue: '#5F9EA0',
  steelblue: '#4682B4',
  cornflowerblue: '#6495ED',
  deepskyblue: '#00BFFF',
  dodgerblue: '#1E90FF',
  lightblue: '#ADD8E6',
  skyblue: '#87CEEB',
  lightskyblue: '#87CEFA',
  midnightblue: '#191970',
  darkblue: '#00008B',
  mediumblue: '#0000CD',
  royalblue: '#4169E1',
  blueviolet: '#8A2BE2',
  indigo: '#4B0082',
  darkslateblue: '#483D8B',
  slateblue: '#6A5ACD',
  mediumslateblue: '#7B68EE',
  mediumpurple: '#9370DB',
  darkmagenta: '#8B008B',
  darkviolet: '#9400D3',
  darkorchid: '#9932CC',
  mediumorchid: '#BA55D3',
  thistle: '#D8BFD8',
  plum: '#DDA0DD',
  violet: '#EE82EE',
  fuchsia: '#FF00FF',
  orchid: '#DA70D6',
  mediumvioletred: '#C71585',
  palevioletred: '#DB7093',
  deeppink: '#FF1493',
  hotpink: '#FF69B4',
  lightpink: '#FFB6C1',
  pink: '#FFC0CB',
  antiquewhite: '#FAEBD7',
  beige: '#F5F5DC',
  bisque: '#FFE4C4',
  blanchedalmond: '#FFEBCD',
  wheat: '#F5DEB3',
  cornsilk: '#FFF8DC',
  lemonchiffon: '#FFFACD',
  lightgoldenrodyellow: '#FAFAD2',
  lightyellow: '#FFFFE0',
  saddlebrown: '#8B4513',
  sienna: '#A0522D',
  chocolate: '#D2691E',
  peru: '#CD853F',
  sandybrown: '#F4A460',
  burlywood: '#DEB887',
  tan: '#D2B48C',
  rosybrown: '#BC8F8F',
  moccasin: '#FFE4B5',
  navajowhite: '#FFDEAD',
  peachpuff: '#FFDAB9',
  mistyrose: '#FFE4E1',
  lavenderblush: '#FFF0F5',
  linen: '#FAF0E6',
  oldlace: '#FDF5E6',
  papayawhip: '#FFEFD5',
  seashell: '#FFF5EE',
  mintcream: '#F5FFFA',
  slategray: '#708090',
  lightslategray: '#778899',
  lightsteelblue: '#B0C4DE',
  lavender: '#E6E6FA',
  floralwhite: '#FFFAF0',
  aliceblue: '#F0F8FF',
  ghostwhite: '#F8F8FF',
  honeydew: '#F0FFF0',
  ivory: '#FFFFF0',
  azure: '#F0FFFF',
  snow: '#FFFAFA',
  dimgray: '#696969',
  darkgray: '#A9A9A9',
  lightgray: '#D3D3D3',
  gainsboro: '#DCDCDC',
  whitesmoke: '#F5F5F5'
}

const colorSet: IColorSet = {
  base: {
    background: colors.black,
    foreground: colors.white,
    color1: colors.darkorange,
    color2: colors.darkorchid,
    color3: colors.mediumspringgreen,
    color4: colors.indianred
  }
};

generateTheme(themeName, colorSet, path.join(__dirname, 'theme.json'));
